tests/unit/test_footprinter.cc

```diff
@@ -543,4 +555,61 @@ TEST_F(TestFootprinter, test_virtual_serialize) {
   }
 }
 
+TEST_F(TestFootprinter, test_ompi) {
+  {
+    std::vector<ompi_communicator_t*> v(3);
+
+    EXPECT_EQ(
+      checkpoint::getMemoryFootprint(v),
+      sizeof(v) + v.capacity() * sizeof(ompi_communicator_t*)
+    );
+  }
+
+  {
+    std::vector<ompi_group_t*> v(5);
+
+    EXPECT_EQ(
+      checkpoint::getMemoryFootprint(v),
+      sizeof(v) + v.capacity() * sizeof(ompi_group_t*)
+    );
+  }
+
+  {
+    std::vector<ompi_request_t*> v(7);
+
+    EXPECT_EQ(
+      checkpoint::getMemoryFootprint(v),
+      sizeof(v) + v.capacity() * sizeof(ompi_request_t*)
+    );
+  }
+
+  {
+    std::vector<ompi_win_t*> v(9);
+
+    EXPECT_EQ(
+      checkpoint::getMemoryFootprint(v),
+      sizeof(v) + v.capacity() * sizeof(ompi_win_t*)
+    );
+  }
+}
+
+struct TestNoSerialize {
+  double d;
+  int i;
+};
+
+TEST_F(TestFootprinter, test_no_serialize) {
+  std::vector<TestNoSerialize> v(7);
+
+  std::cout << std::boolalpha;
+  std::cout << SerializableTraits<int>::is_footprintable_v << std::endl;
+  std::cout << SerializableTraits<Test1>::is_footprintable_v << std::endl;
+  std::cout << SerializableTraits<TestNoSerialize>::is_footprintable_v << std::endl;
```

Comment:
no success with the generalized approach, I've just pushed a somewhat cleaned up current state and I'm throwing in the towel for today

- I have moved the detection into `dispatch::serializeArray`
- `is_footprintable` in current form doesn't work as intended - here, calling it on `TestNoSerialize` class yields `true`
- I have tried to use somewhat along the lines of
```cpp
not (SerializableTraits<T>::has_serialize_function or
  SerializableTraits<T>::is_bytecopyable)
```
but couldn't get it to work either (dealing with redefinition, ambiguous call, template deduction errors)

I have implemented serializer for `std::list::iterator` in the meantime, but this came with its own issues (see comment in `list_serialize.h`).

Reply:
@PhilMiller @lifflander I would appreciate any help with creating the general solution (a footprinting serializer for containers whose elements are not serializable, in *vt* we have such case with `std::vector` and `std::unordered_map`)